crates/engine/tree/src/tree/mod.rs

```diff
@@ -40,14 +40,22 @@ use std::{
     marker::PhantomData,
     sync::{mpsc::Receiver, Arc},
 };
-use tokio::sync::{mpsc::UnboundedSender, oneshot};
+use tokio::sync::{
+    mpsc::{UnboundedReceiver, UnboundedSender},
+    oneshot,
+};
 use tracing::*;
 
 mod memory_overlay;
 pub use memory_overlay::MemoryOverlayStateProvider;
 
 /// Maximum number of blocks to be kept only in memory without triggering persistence.
 const PERSISTENCE_THRESHOLD: u64 = 256;
+/// Number of pending blocks that cannot be executed due to missing parent and
+/// are kept in cache.
+const DEFAULT_BLOCK_BUFFER_LIMIT: u32 = 256;
```

Comment:
These constants are defined to be able to instantiate `EngineApiTreeState` when building the tree, lmk wdyt

Reply:
I think this makes sense, given how the block buffer works rn